test/printingTools.js

```diff
@@ -0,0 +1,409 @@
+// This test can be run directly using mocha:
+// $ npx mocha test/printingTools.js
+// Using truffle test works but it's much slower, since it needs to compile contracts and deploy them to the blockchain.
+
+const assert = require("assert")
+const BN = require("bn.js")
+
+const { toErc20Units, fromErc20Units, bnMaxUint, bnOne } = require("../scripts/utils/printing_tools")
+
+const goodTwoWayPairs = [
+  {
+    user: "1.1",
+    machine: "1100",
+    decimals: 3,
+  },
+  {
+    user: "0.01",
+    machine: "100",
+    decimals: 4,
+  },
+  {
+    user: "1",
+    machine: "100",
+    decimals: 2,
+  },
+  {
+    user: "104",
+    machine: "104",
+    decimals: 0,
+  },
+  {
+    user: "0.002901",
+    machine: "2901000000000000",
+    decimals: 18,
+  },
+  {
+    user: "1.002901",
+    machine: "1002901000000000000",
+    decimals: 18,
+  },
+  {
+    user: "0." + bnMaxUint.toString().padStart(255, "0"),
+    machine: bnMaxUint.toString(),
+    decimals: 255,
+  },
+  {
+    user: bnMaxUint.toString(),
+    machine: bnMaxUint.toString(),
+    decimals: 0,
+  },
+  {
+    user: bnMaxUint.toString().slice(0, -18) + "." + bnMaxUint.toString().slice(-18),
+    machine: bnMaxUint.toString(),
+    decimals: 18,
+  },
+  {
+    user: "0",
+    machine: "0",
+    decimals: 0,
+  },
+  {
+    user: "0",
+    machine: "0",
+    decimals: 18,
+  },
+  {
+    user: "0",
+    machine: "0",
+    decimals: 255,
+  },
+  {
+    user: "11." + "".padEnd(30, "1"),
+    machine: "".padEnd(32, "1"),
+    decimals: 30,
+  },
+  {
+    user: "0.0" + "".padEnd(29, "1"),
+    machine: "".padEnd(29, "1"),
+    decimals: 30,
+  },
+  {
+    user: "".padEnd(30, "1"),
+    machine: "".padEnd(30, "1"),
+    decimals: 0,
+  },
+  {
+    user: "222.222222222222222002",
+    machine: "222222222222222222002",
+    decimals: 18,
+  },
+]
+
+const invalidDecimals = function(decimals) {
+  return "Invalid number of decimals for ERC20 token: " + decimals.toString()
+}
+const tooLargeNumber = function() {
+  return "Number larger than ERC20 token maximum amount (uint256)"
+}
+
+describe("fromUserToMachineReadable", () => {
+  const invalidNumber = function(amount) {
```

Comment:
Can we put all the error messages outside of the describe block? Alternatively, you could put them all inside...